kotlin-inject-compiler/ksp/src/main/kotlin/me/tatarka/inject/compiler/ksp/InjectProcessor.kt

```diff
@@ -42,9 +42,17 @@ class InjectProcessor(
 
         val componentSymbols =
             resolver.getSymbolsWithAnnotation(COMPONENT.canonicalName).filterIsInstance<KSClassDeclaration>()
-        val deferredClasses = componentSymbols.filterNot { element ->
-            processInject(element, provider, codeGenerator, injectGenerator)
-        }.toList()
+        val deferredClasses = componentSymbols
+            .distinctBy {
+                // Classes keep being processed multiple times and throw a FileAlreadyExistsException in multi round
+                // scenarios. The function above getSymbolsWithAnnotation returns deferred symbols twice for some
+                // reason in the next round.
+                it.qualifiedName
```

Comment:
Simple enough, I assume a ksp issue is filed for it? Would be nice to include in the comment here

Reply:
Done https://github.com/google/ksp/issues/1993